common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -460,6 +462,70 @@ protected TokenResult performTokenRequest(@SuppressWarnings(WarningType.rawtype_
         return tokenResult;
     }
 
+    /**
+     * Renewing AT of nested app.
+     */
+    protected synchronized void renewAccessTokenForNestedApp(@NonNull final SilentTokenCommandParameters parameters,
```

Comment:
>renewAccessTokenForNestedApp [](http://example.com/codeflow?start=32&length=28)

Why in BaseController? would non broker controller need this? 
This method by itself has no check around NAA or FOCI, so can be called erroneously. 

I think Foci RT request logic can be directly inside a BrokerLocalController private method. Then you can call renewAccessToken()

Reply:
agreed. will try to move this to BrokerLocalController and also add a check for it to not be used erroneously. This is not required in any other cases.